src/services/ActivityService.ts

```diff
@@ -84,39 +84,33 @@ export class ActivityService {
 
     /**
      * Ends an activity with the given id
+     * if the visit was already closed, the response will be 200 and the flag visitClosed will be set to true
      * @param id - id of the activity to end
-     * @returns Promise void
+     * @returns Promise<{visitClosed: boolean}>
      */
-    public async endActivity(id: string): Promise<void> {
-        return this.dbClient.get({ id })
-            .then(async (result: DocumentClient.GetItemOutput): Promise<void> => {
-                // Result checks
-                if (result.Item === undefined) {
-                    throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
-                }
+    public async endActivity(id: string): Promise<{ visitClosed: boolean }> {
+        try {
 
-                if (result.Item.endTime !== null) {
-                    throw new HTTPResponse(403, { error: HTTPRESPONSE.ALREADY_ENDED });
-                }
+            const result: DocumentClient.GetItemOutput = await this.dbClient.get({ id });
 
-                const activity: IActivity = result.Item as IActivity;
-
-                // Assign the endTime
-                const endTime: string = new Date().toISOString();
-                Object.assign(activity, { endTime });
+            if (result.Item === undefined) {
+                console.log(`Error occurred: ${HTTPRESPONSE.NOT_EXIST} with statusCode: 404`);
+                throw new HTTPResponse(404, { error: HTTPRESPONSE.NOT_EXIST });
+            }
 
+            // visit is opened as there is no end time and we are not closing it so we record an endTime
+            if (result.Item.endTime === null) {
+                console.log(`Visit with ID ${id} is already closed`);
+                const activity: IActivity = result.Item as IActivity;
+                activity.endTime = new Date().toISOString();
                 await this.dbClient.put(activity);
-            })
-            .catch((error: AWSError | HTTPResponse) => {
-                // If we get HTTPResponse, we rethrow it
-                if (error instanceof HTTPResponse) {
-                    console.error(`Error occurred: ${error.body}`);
-                    throw error;
-                }
+            }
 
-                // Otherwise, if DynamoDB errors, we throw 500
-                throw new HTTPResponse(error.statusCode, { error: `${error.code}: ${error.message} At: ${error.hostname} - ${error.region} Request id: ${error.requestId}` });
-            });
+            return { visitClosed: true };
+
+        } catch (error) {
+            throw new HTTPResponse(error.statusCode, { error: `${error.code}: ${error.message} At: ${error.hostname} - ${error.region} Request id: ${error.requestId}` });
```

Comment:
@bristol-mike-jones removed the catch block since we are throwing a response in the try block and refactored to log the message once the visitId is not found